tests/ZonedDateTimeTest.php

```diff
@@ -482,4 +486,438 @@ public function providerCompareTo() : array
+
+        $this->assertDayOfWeekIs(4, $zonedDateTime->getDayOfWeek());
+    }
+
+    public function testGetDayOfYear()
+    {
+        $timeZone = TimeZone::parse('America/Los_Angeles');
+        $localDateTime = '2000-01-20T12:34:56.123456789';
+        $localDateTime = LocalDateTime::parse($localDateTime);
+        $zonedDateTime = ZonedDateTime::of($localDateTime, $timeZone);
+
+        $this->assertSame(20, $zonedDateTime->getDayOfYear());
+    }
+
+    public function testGetHour()
+    {
+        $timeZone = TimeZone::parse('America/Los_Angeles');
+        $localDateTime = '2000-01-20T12:34:56.123456789';
+        $localDateTime = LocalDateTime::parse($localDateTime);
+        $zonedDateTime = ZonedDateTime::of($localDateTime, $timeZone);
+
+        $this->assertSame(12, $zonedDateTime->getHour());
+    }
+
+    public function testGetMinute()
+    {
+        $timeZone = TimeZone::parse('America/Los_Angeles');
+        $localDateTime = '2000-01-20T12:34:56.123456789';
+        $localDateTime = LocalDateTime::parse($localDateTime);
+        $zonedDateTime = ZonedDateTime::of($localDateTime, $timeZone);
+
+        $this->assertSame(34, $zonedDateTime->getMinute());
+    }
+
+    public function testGetSecond()
+    {
+        $timeZone = TimeZone::parse('America/Los_Angeles');
+        $localDateTime = '2000-01-20T12:34:56.123456789';
+        $localDateTime = LocalDateTime::parse($localDateTime);
+        $zonedDateTime = ZonedDateTime::of($localDateTime, $timeZone);
+
+        $this->assertSame(56, $zonedDateTime->getSecond());
+    }
+
+    public function testWithDate()
+    {
+        $y = (int)date('Y');
+        $m = (int)date('m');
+        $d = (int)date('d');
+        $timeZone = TimeZone::parse('America/Los_Angeles');
+        $localDateTime = '2000-01-20T12:34:56.123456789';
+        $localDateTime = LocalDateTime::parse($localDateTime);
+        $newDate = LocalDate::of($y, $m, $d);
+        $zonedDateTime = ZonedDateTime::of($localDateTime, $timeZone);
+
+        $this->assertSame(date('Y-m-d'), (string)$zonedDateTime->withDate($newDate)->getDate());
+    }
+
+    public function testWithTime()
+    {
+        $timeZone = TimeZone::parse('America/Los_Angeles');
+        $localDateTime = '2000-01-20T12:34:56.123456789';
+        $localDateTime = LocalDateTime::parse($localDateTime);
+        $time = LocalTime::of(12, 34, 56, 987654321);
+        $zonedDateTime = ZonedDateTime::of($localDateTime, $timeZone);
+
+        $this->assertSame('12:34:56.987654321', (string)$zonedDateTime->withTime($time)->getTime());
+    }
+
+    public function testWithYear()
+    {
+        $timeZone = TimeZone::parse('America/Los_Angeles');
+        $localDateTime = '2000-01-20T12:34:56.123456789';
+        $localDateTime = LocalDateTime::parse($localDateTime);
+        $zonedDateTime = ZonedDateTime::of($localDateTime, $timeZone);
+
+        $this->assertSame(2000, $zonedDateTime->withYear(2000)->getYear());
```

Comment:
Please check the whole zoned date-time string as well:
`$this->assertSame('...', (string) $zonedDateTime->withYear(...))`

Is this on purpose that you're setting the same year as the one in the original date? Not sure which part of the code you're testing here. If not, it'd be actually more useful to test with another year (or perform both tests, same year and other year).

Same for `withMonth()`, `withDay()`, and others below.

Reply:
I do the same things for the ```withHour```, ```withMinute```, ```withSecond```, ```withNano``` and ```withFixedOffsetTimeZone```.